client/client-core/src/main/java/com/cognifide/aet/common/TestSuiteRunner.java

```diff
@@ -60,79 +45,43 @@ protected DateFormat initialValue() {
-                         String buildDirectory, long timeout, boolean xUnit) throws JMSException, IOException {
+  public TestSuiteRunner(String endpointDomain, String buildDirectory, int timeout, String domain,
+      boolean xUnit) {
     this.redirectWriter = new RedirectWriter(buildDirectory);
     this.buildDirectory = buildDirectory;
     this.timeout = timeout;
     this.endpointDomain = endpointDomain;
+    this.domain = domain;
     this.xUnit = xUnit;
-    CommunicationSettings communicationSettings = getSettings();
-    reportDomain = communicationSettings.getReportDomain();
-    JmsEndpointConfig jmsEndpointConfig = communicationSettings.getJmsEndpointConfig();
-    ActiveMQConnectionFactory activeMQConnectionFactory = new ActiveMQConnectionFactory(jmsEndpointConfig.getUser(),
-            jmsEndpointConfig.getPass(), jmsEndpointConfig.getUri());
-    activeMQConnectionFactory.setTrustAllPackages(true);
-    connection = activeMQConnectionFactory.createConnection();
-    session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
-    producer = session.createProducer(session.createQueue(inQueueName));
-    outRunnerDestination = session.createTemporaryQueue();
-    connection.start();
+    suiteExecutionResponseHandler = new JSONResponseHandler<>(SuiteExecutionResult.class);
+    suiteStatusResponseHandler = new JSONResponseHandler<>(SuiteStatusResult.class);
   }
 
-  private CommunicationSettings getSettings() throws IOException {
-    try {
-      URL configUrl = new URL(endpointDomain + "/configs/communicationSettings");
-      String communicationSettings = IOUtils.toString(configUrl.openStream());
-      return new Gson().fromJson(communicationSettings, CommunicationSettings.class);
-    } catch (IOException e) {
-      throw new IOException("Unable to get Configs form AET Configs service", e);
-    }
-  }
-
-  public void runTestSuite(final TestSuiteRun testSuiteRun) throws AETException {
+  public void runTestSuite(final File testSuite) throws AETException {
     RunnerTerminator runnerTerminator = new RunnerTerminator();
     try {
-      final Suite suite = new SuiteFactory().suiteFromTestSuiteRun(testSuiteRun);
-      suite.validate(Sets.newHashSet("version", "runTimestamp"));
-      new LockClient(suite, endpointDomain, runnerTerminator).tryToSetLock();
-      TaskMessage taskMessage = new TaskMessage<>(MessageType.RUN, suite);
-      ObjectMessage message = session.createObjectMessage(taskMessage);
-      message.setJMSReplyTo(outRunnerDestination);
-      consumer = session.createConsumer(outRunnerDestination);
-      producer.send(message);
+      SuiteExecutionResult suiteExecutionResult = startSuiteExecution(testSuite);
 
       String now = DATE_FORMATTER.get().format(new Date());
-      Logger.info(this, String.format("CorrelationID: %s", testSuiteRun.getCorrelationId()));
+      Logger.info(this, String.format("CorrelationID: %s", suiteExecutionResult.getCorrelationId()));
```

Comment:
it's easier to use varargs method signature - http://log.jcabi.com/apidocs-0.12.2/com/jcabi/log/Logger.html#info(java.lang.Object, java.lang.String, java.lang.Object...). 
This way the formatting won't happen if log level setting hides the message